### ISSUE TITLE
set MAVEN mode as the default mode

### ISSUE THREAD

**Issue by monperrus** [SpoonLabs/sorald#681]
follow up of https://github.com/SpoonLabs/sorald/pull/347

MAVEN mode should give better bindings, better imports and overall better results.

**Comment by algomaster99** [comment]
@monperrus Using the maven mode requires passing `--resolve-classpath-from` and the client has to give it the path to a valid maven project. I think we should not make it the default mode because if someone does not want to use the above flag, they will have to explicitly specify that they want the current default mode.

Maybe we can highlight in the `README` that we recommend using maven mode. What do you think?

**Comment by monperrus** [comment]
Good point. In Maven mode, regardless it's default or not, we can automatically infer the arguments  to be given to `--resolve-classpath-from`

**Comment by algomaster99** [comment]
More context: https://github.com/SpoonLabs/sorald/issues/598#issuecomment-1008903095.
